### PR TITLE
Port coordinates module

### LINKED ISSUE #884: Migrate from nose to py.test
See: https://nose.readthedocs.io/en/latest/

Alternatives are py.test and nose2

http://nose2.readthedocs.io/en/latest/

http://pytest.org/latest/

I've only have additional experience with py.test and actually like it very much.

This should be done in one go as test-fixtures are handled differently in py.test for example.

### PR DESCRIPTION
**Pull request by utkbansal**
As discussed, this is some ancient work and might involve a messy rebase. 

This was the status back then:
PyTest - 1408 passed, 68 skipped
Nose - Ran 1675 tests in 172.808s OK (SKIP=64)

Partially Fixes #884 

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: testsuite/MDAnalysisTests/coordinates/test_xyz.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 from MDAnalysisTests.coordinates.base import (BaseReaderTest, BaseReference,
                                               BaseWriterTest)
 from MDAnalysisTests import tempdir, make_Universe
-
+from numpy.testing import TestCase
```
Comment (richardjgowers) [inline testsuite/MDAnalysisTests/coordinates/test_xyz.py]: is `numpy.testing.TestCase` the same as `unittest.TestCase`?

Comment (utkbansal) [inline testsuite/MDAnalysisTests/coordinates/test_xyz.py]: Looks like it - https://github.com/numpy/numpy/blob/master/numpy/testing/__init__.py#L10
Should be able to just use `unittest.TestCase` in its place.

Comment (kain88-de) [inline testsuite/MDAnalysisTests/coordinates/test_xyz.py]: Pleas use unittest.TestCase consistently then.

**Comment by utkbansal** [comment]
It's interesting that the minimal build passes but the full build fails. From pytest's point of view, both should be the same.

**Comment by utkbansal** [comment]
This should fix the error, but I expect coverage to drop.

**Comment by kain88-de** [comment]
@utkbansal you should be able to see the combined coverage report off-line on your laptop as well. A good idea is also the use a coverage report (html best) of just the coordinates module from the develop branch and use that as a reference. Doing all of these tests locally will be much faster then waiting for travis.

**Comment by utkbansal** [comment]
Something fishy going on with coverage. 🤔

**Comment by kain88-de** [comment]
what do your offline tests show for the coverage?

**Comment by utkbansal** [comment]
My system shows a drop in coverage, which I was expecting. The problem with
coveralls is that it is passing the check even though the coverage has
dropped.

**Comment by kain88-de** [comment]
Can you fix the coverage drop locally so you have the same pytest coverage as your nose reference?

**Comment by utkbansal** [comment]
Yes, I'm working on it. But I would feel better if coveralls worked.

**Comment by kain88-de** [comment]
That is not the first time coveralls has such hiccups. I'm not sure what is the reason for this. But we can still use the numbers to check for large drops in coverage.

**Comment by jbarnoud** [comment]
Avoid  "Partially Fixes" in PR descriptions: github does not know about "partially" and closes the referenced issue.

**Comment by utkbansal** [comment]
Woah, this wasn't supposed to be merged yet!

**Comment by jbarnoud** [comment]
@utkbansal Do you think it hurts that it is merged? From what I see I do not think so, but maybe I am missing something. If not, let's just keep this here and you can continue your changes from there.

In the future, prefix the title of your PR with "[WIP]" to make clear it is not ready. If you can add label to PRs, you can also attach the "Work in progress" label. It is a bright right tag that should warn us that we should proceed with caution.

**Comment by utkbansal** [comment]
@jbarnoud There is a major drop in coverage!

**Comment by richardjgowers** [comment]
Ah sorry, it had green lights. Open an issue for the drop in coverage

**Pull request merged**